src/Sarif.FunctionalTests/v2/ConverterTestData/PREfast/Expected-022.xml.sarif

```diff
@@ -12,46 +12,55 @@
           "mimeType": "text/x-cpp"
         }
       },
-      "logicalLocations": {
-        "f": {
+      "logicalLocations": [
+        {
+          "fullyQualifiedName": "f",
```

Comment:
>fullyQualifiedName [](start = 11, length = 18)

FWIW, the `name` property is required. `fullyQualifiedName` is required for non-top-level locations, optional for top-level locations (because then it is the same as `name`). See §3.27.3,  §3.27.4.

Rationale:
1. Before the dictionary-to-array transformation, you always had FQLN available because it was the dictionary key. We require the `fQLN` property, now, to get parity with that functionality.
2. Without `fQLN` being present, a consumer would have to (a) walk the parent tree to get the sequence of components that comprise the FQLN, and (b) know the language-specific syntax for composing the components into the FQLN string.
3. `name` is required because without it, a consumer would have to know the language-specific way to parse it out of the FQLN.

Reply:
TODO: [https://github.com/Microsoft/sarif-sdk/issues/1173](https://github.com/Microsoft/sarif-sdk/issues/1173)

---
In reply to: [243851854](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243851854) [](ancestors = 243851854)